assets/template/src/lib.rs

```diff
@@ -15,6 +15,7 @@ blueprint! {
             let my_bucket: Bucket = ResourceBuilder::new_fungible()
                 .metadata("name", "HelloToken")
                 .metadata("symbol", "HT")
+                .auth("take_from_vault", auth!(allow_all))
```

Comment:
Feel like having an enum for all the resource methods would be nice, since they're finite and predefined.

Reply:
I'm actually thinking of taking this a step further and implement something like:
```
.take_auth(auth!(allow_all))
.mint_auth(auth!(require()))
```
Thoughts?

Reply:
I personally like `auth()` more. We will likely reuse the enum in other places, like `update_auth()`. Don't want to have too many `update_xxx_auth()` methods in the library.